### ISSUE TITLE
[feature request]unset old catagories

### ISSUE THREAD

**Issue by NeonLightning** [rallion/depressurizer#57]
i'm hoping for a way to unset catagories after they are no longer valid. i had catagories for ratings set for ever 10% rating. and then i noticed i had alot in 90 so i made that only valid to 94 instead of 99 and made one from 95-99 but all the ones that were set to 90 before are still set to 90 and now also any that are 95+ are in that too.

**Comment by Theo47** [comment]
The easiest way to do so would be to delete the "90" category. That would unset that category from all games that have it.

Afterwards you just have to autocategorize those games again.

if you use [this version](https://github.com/Theo47/depressurizer/releases/tag/v0.7.1.1) instead then you can use manual autocat and group autocat to automate the above process, effectively implementing the feature you requested yourself.

**Comment by NeonLightning** [comment]
i ended up just unsetting all my catagories from everything then autocat'ing again.
also sorry but i'm not exactly sure what you mean by using manual autocat and group autocat to automate that.
since i so recently did that i'm not able to confirm right now. but another example of this is keeping the "Early Access" catagory updated. i seem to recall untill i cleared all settings and reset them it had several games that were no longer in.
with a library of over 700 games its hard to keep track of what games i need to manually update.

**Comment by Theo47** [comment]
Well what you did was exactly what i was suggesting. Unset relevant categories from all the games you want to autocat then autocat those games instead.

For "early access" you would want to remove the "early access" category from all games before you autocat those games again.

**Comment by NeonLightning** [comment]
basically this is a workaround method. and to unset all categories takes a good 10-15 or more minutes of a frozen program with this many games in my list(on a lower end gaming pc)

**Comment by Theo47** [comment]
Well again that was exactly what i was trying to suggest. A workaround method.

It will probably take a while for this to be properly implemented so i thought a workaround method would be useful to you in the meantime.

I am suprised it takes 10-15 mins for you though. On my 6-year-old pc with 1000+ games it only takes a few seconds.

**Comment by NeonLightning** [comment]
appologies i realize my statement sounded kinda jerkish. but anywho i have a fair few categories.and each time i select all and check or uncheck a category it locks up for about 10-25 seconds and if i click more categories while its locked up it'll usually process all of them eventually but for each one i click while its locked up it seems to add a bit more time ontop of that.10-25 per category.

**Comment by Theo47** [comment]
I wouldn't say it sounded jerkish. To me it just sounded like there was there was a misunderstanding somewhere. Especially with the first part of the conversation that sounded to me like:
"-You should unset all categories."
"-I didn't do what you said. Instead i unsetted all categories."

Now back to the problem at hand i see what you mean. Apparently we were removing the categories by using 2 different ways. Doing it your way takes several minutes on my pc too.

Try doing it this way instead. Ctrl-select all the categories you want then press the delete button. Like in this [pic](http://imgur.com/4sQihMZ).

Alternatively you can use the manual autocat i suggested earlier. [This pic](http://imgur.com/M71sVLs) should explain what i meant by that. Keep in mind that you need to use the version i linked earlier for that.

p.s: Turns out i was right about the misunderstanding. We were removing the categories in different ways, with it taking 10 mins for you and 10 secs for me which made us see the problem differently.

**Comment by NeonLightning** [comment]
yep that's exactly the misunderstanding. thanks that actually does work fairly well.
in my mind deleting the categories would mess up the autocats(i realize how silly that is)
so setup a autocat that removes all(didn't realize that was an option) set it to the top of the autocat list and honestly that does exactly what i need. i can see how it might cause problems for some people but does exactly what i need.
idk if/how this would work on rallion's one as i'm using your fork now but it works so i'm happy.

_edit_
though i have noticed that if you add a category for ALL and set the manual remove to remove any cats aside from all it still has a heavy delay

**Comment by Theo47** [comment]
Deleting categories won't mess up the autocats (or if it actually does then it is a bug and should be reported)

> though i have noticed that if you add a category for ALL and set the manual remove to remove any cats aside from all it still has a heavy delay

I had no idea this happened. I'll see if i can figure out what is causing the delay and how to fix it but it will probably take a while.

In the meantime i am going to propose another workaround.
Instead of pressing ctrl-a and selecting all 700 of your games, you should just select the <ALL> category instead. Also uncheck the Selected only checkbox below the autocategorize button. This should improve speed.
